### PR TITLE
Expanded model tests for non reference files.

### PR DESCRIPTION
**Pull request by PaulHuwe**
Changes:
- Updated model subclass code - changed from returning a generator to a set for use with more complicated model selections.
- Corrected time format in tests to astropy time objects.
- Cleaned up old tests to better reflect present models.
- Added check for core metadata inclusion in nonreference files.

Ticket:
- RCAL-88

Tests:
- This is a test file.

Documentation:
- Not implemented.

### CODE REVIEW

**Review by eslavich — COMMENTED** [review]
The other test that I thought would be useful here is to check that the schema_url attribute refers to a schema that actually exists.  You can check that by calling `asdf.schema.load_schema(model.schema_url)`.

File: romancal/datamodels/tests/test_models.py [hunk 1]
```diff
@@ -120,6 +129,15 @@ def test_reference_file_model(tmp_path, model_class):
     assert_referenced_schema(model_class.schema_url,
                              datamodels.reference_files.referencefile.ReferenceFileModel.schema_url)
 
+# Common tests for nonReference Models
+@pytest.mark.parametrize("model_class", list(iter_subclasses(
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/datamodels/tests/test_models.py]: Schema URL is already teat in lines 24-26.

Other suggested changes implemented.

**Review by ddavis-stsci — CHANGES_REQUESTED** [review]
You'll need to update the changes file. Looking at it I must have overwritten or messed up the CHANGES.rst from the release. If you could add to the top of the changes file,

```
0.x.0 (unreleased)
==================

datamodels
----------

- Your test text [#169]
```

**Comment by PaulHuwe** [comment]
Change log updated.

**Review by ddavis-stsci — CHANGES_REQUESTED** [review]
Looks good. One last thing is that the Changes.rst should also contain the github issue in brackets, so 
`- blah blah blah [#169]`
so that we can track code changes to the issue.

**Review by ddavis-stsci — APPROVED** [review]
LGTM

**Pull request merged**